clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -46,7 +49,16 @@
  * by the default implementation.
  */
 @PublicApi
-interface FileSystem {
+public interface FileSystem {
+  class Factory {
+    public static FileSystem create() {
+      if (ClientContext.getConf().getBoolean(Constants.USER_LINEAGE_ENABLED)) {
+        return LineageFileSystem.get();
+      }
+      return new BaseFileSystem();
```

Comment:
Good point, I've changed both to use `get()`. I changed the constructor behavior from a singleton to new instances since there currently is no state in the FileSystems.